c2corg_api/views/association.py

```diff
@@ -76,6 +78,9 @@ def collection_delete(self):
             if association is None:
                 raise HTTPBadRequest('association does not exist')
 
+        if is_main_waypoint_association(association):
+            raise HTTPBadRequest('waypoint is still main waypoint of route')
```

Comment:
Yes, that makes it a bit clearer. But remember that the error messages that the API returns are not intended to be shown directly to the user.